src/translator/sentence_ranges.h

```diff
@@ -1,52 +1,253 @@
+    assert(flatIdx < flatByteRanges_.size());
+    return flatByteRanges_[flatIdx];
+  }
+
+  /// Const reference required for use in the conversion function.
+  const std::vector<string_view_type> &flatByteRanges() const {
+    return flatByteRanges_;
+  }
+
+  /// Const reference required for use in the conversion function.
+  const std::vector<size_t> &sentenceBeginIds() const {
+    return sentenceBeginIds_;
+  }
 
 private:
-  // A flat storage for string_views. Can be words or sentences.
-  std::vector<string_view> flatByteRanges_;
+  /// A flat storage for string_view_types. Can be words or sentences.
+  std::vector<string_view_type> flatByteRanges_;
 
-  // The container grows dynamically with addSentence. size_t marking index is
-  // used to ensure the sentence boundaries stay same while underlying storage
-  // might be changed during reallocation.
+  /// The container grows dynamically with addSentence. size_t marking index is
+  /// used to ensure the sentence boundaries stay same while underlying storage
+  /// might be changed during reallocation.
   std::vector<size_t> sentenceBeginIds_;
 
-  // Utility function to extract the string starting at firstWord and ending at
-  // lastWord as a single string-view.
-  string_view sentenceBetween(string_view firstWord,
-                              string_view lastWord) const;
+  /// Utility function to extract the string starting at firstWord and ending at
+  /// lastWord as a single string-view.
+  string_view_type sentenceBetween(string_view_type firstWord,
+                                   string_view_type lastWord) const {
+
+    const char *data = firstWord.data();
+    size_t size = lastWord.data() + lastWord.size() - firstWord.data();
+    return string_view_type(data, size);
+  };
 };
 
-} // namespace bergamot
+// This typedef allows a non-invasive modification (as SentenceRanges are
+// present in many files which operate during the lifecycle of a Request).
+typedef AnnotationT<string_view> SentenceRanges;
+
+/// An AnnotatedBlob is blob std::string along with the annotation which are
+/// valid until the original blob is valid (string shouldn't be invalidated).
+/// Annotated Blob, due to its nature binding a blob to string_view should only
+/// be move-constructible as a whole.
+
+template <class annotation_type> struct AnnotatedBlobT {
+public:
+  std::string blob;
+  AnnotationT<annotation_type> annotation;
 
+  AnnotatedBlobT(){};
+
+  /// Proper constructor.
+  AnnotatedBlobT(std::string &&blob, AnnotationT<annotation_type> &&annotation)
+      : blob(std::move(blob)), annotation(std::move(annotation)) {}
+
+  /// An improper constructor. Annotated Blobs "can" also be converted
+  /// between string_views. This templated constructor allows for the same.
+  template <class other_annotation_type>
+  AnnotatedBlobT(std::string &&blob,
+                 const AnnotationT<other_annotation_type> &annotation)
+      : blob(std::move(blob)), // string is moved.
```

Comment:
Kill this and just use absl::string_view everywhere.  @abhi-agg already said `std::string_view` doesn't work for him, so let's stop converting stuff.

Reply:
https://github.com/mozilla/bergamot-translator/issues/8#issuecomment-800163279

Apparently pair of size_t offsets are the WASM desired primitive. absl::string_view is still expected to be remain in ssplit-cpp (cc @ugermann) and we'll probably be keeping it in our marian sentencepiece modifications as well. Given the overall hate for conversions and two structs doing the same thing, permission to keep ByteRange everywhere (including marian replacement server)?

```
ByteRange {
    size_t begin_byte_offset;
    size_t end_byte_offset;
}

```

Reply:
Resolved. No more std::string_view. It's the above structure all the way down, inside Annotation as well. string_views are constructed when required using the ranges and the reference string and internally converted to ByteRanges when added using string-views.